src/libraries/Common/src/System/Runtime/InteropServices/ComEventsMethod.cs

```diff
@@ -99,7 +100,7 @@ private void PreProcessSignature()
         /// Since multicast delegate's built-in chaining supports only chaining instances of the same type,
         /// we need to complement this design by using an explicit linked list data structure.
         /// </summary>
-        private readonly List<DelegateWrapper> _delegateWrappers = new List<DelegateWrapper>();
+        private List<DelegateWrapper> _delegateWrappers = new List<DelegateWrapper>();
```

Comment:
It seems like optimization of this code / reduction in allocation isn't needed.  But if we decide to in the future, there are various things that could be done.  For example, this empty list could be a singleton, or null could be special-cased to mean empty.  And if one handler is common, rather than being a list this could be an object, which could then store either a DelegateWrapper directly or a list of them.

Reply:
Switched back to arrays and am now using `Array.Empty<T>` for this. It isn't non-allocating, but it is now a single allocation for this code path for process lifetime.